master/buildbot/steps/artifact.py

```diff
@@ -402,6 +472,113 @@ def start(self):
         self.setCommand(command)
         ShellCommand.start(self)
 
+    @defer.inlineCallbacks
+    def _getBuildRequest(self):
+        triggeredbybrid = self.build.requests[0].id
+        br = yield self.master.db.buildrequests.getBuildRequestTriggered(triggeredbybrid, self.artifactBuilderName)
+        defer.returnValue(br)
+
+
+class DownloadArtifactsFromChildren(LoggingBuildStep, CompositeStepMixin):
+    name = "Download Artifact(s) from triggered builds"
+    description="Downloading artifact(s) from triggered builds..."
+    descriptionDone="Downloaded artifact(s) from triggered builds"
+
+    def __init__(self,
+                 artifactServer,
+                 artifactServerDir,
+                 artifactBuilderName,
+                 workDir='',
+                 artifactServerPort=None,
+                 artifactDestination=None,
+                 artifactDirectory=None,
+                 usePowerShell=True,
+                 **kwargs):
+        self.workDir = workDir
+        self.artifactBuilderName = artifactBuilderName
+        self.artifactDirectory = artifactDirectory
+        self.artifactServer = artifactServer
+        self.artifactServerDir = artifactServerDir
+        self.artifactServerPort = artifactServerPort
+        self.artifactDestination = artifactDestination
+        self.master = None
+        self.usePowerShell = usePowerShell
+        LoggingBuildStep.__init__(self, **kwargs)
+
+    @defer.inlineCallbacks
+    def start(self):
+        if self.master is None:
+            self.master = self.build.builder.botmaster.parent
+
+        self._setUpLogs()
+
+        partitionRequests = yield self.master.db.buildrequests.getBuildRequestsTriggeredBy(self.build.requests[0].id, self.artifactBuilderName)
```

Comment:
@mariangemarcano Not sure what you mean? - so are we talking about the child build requests being merged? - the next function `_getBuildRequestIdsWithArtifacts` should sort it out.

If the build request has an `artifactbrid` then we will use that as the real build request otherwise we will use the `brid`. Should the build request be merged at the point of creating the partitions the id's will be duplicated and the duplicates will get removed.

We did some test in the loadtests setup. We can take a look at it tomorrow if you want just so we are sure we haven't misunderstood something.